packages/server/src/error/TRPCError.ts

```diff
@@ -25,28 +24,24 @@ export function getTRPCErrorFromUnknown(cause: unknown): TRPCError {
 }
 
 export class TRPCError extends Error {
-  public readonly cause?;
+  public readonly cause?: Error;
   public readonly code;
 
   constructor(opts: {
     message?: string;
     code: TRPC_ERROR_CODE_KEY;
     cause?: unknown;
   }) {
-    const code = opts.code;
     const message =
-      opts.message ?? getMessageFromUnknownError(opts.cause, code);
-    const cause: Error | undefined =
+      opts.message ?? getMessageFromUnknownError(opts.cause, opts.code);
+    const cause =
       opts.cause !== undefined ? getErrorFromUnknown(opts.cause) : undefined;
 
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore https://github.com/tc39/proposal-error-cause
     super(message, { cause });
```

Comment:
I don't have the history on this, what's the reason for not following the `Error` constructor?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#syntax

Reply:
In what sense?

Reply:
The `Error` constructor accepts a message as the first argument, and then options (or filename and lineNumber), whereas `TRPCError` only accepts an object combining both the message and options. IMO it would be better to follow the same constructor and extend the options to add the `code`, but I'm sure there's a reason why it was made like this (but that would require a breaking change).

Reply:
Ah ok, I don't know the history here as well. From what I see it's most likely because `message` is optional and not always known at compile time. It could feel awkward to do `new TRPCError(null, { code: 420 });`. Just a guess.

Reply:
If there is more that can be done that would result in something breaking we can do it on the `next` branch